src/nnet3/nnet-component-itf.cc

```diff
@@ -96,6 +96,8 @@ Component* Component::NewComponentOfType(const std::string &component_type) {
     ans = new ConvolutionComponent();
   } else if (component_type == "MaxpoolingComponent") {
     ans = new MaxpoolingComponent();
+  } else if (component_type == "Maxpooling3dComponent") {
```

Comment:
OK, fine.
Is maxpooling3d efficient for the (possibly more common) 1d and 2d cases?
Dan

On Sat, Jan 23, 2016 at 6:20 PM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In src/nnet3/nnet-component-itf.cc
> <https://github.com/kaldi-asr/kaldi/pull/460#discussion_r50627193>:
>
> > @@ -96,6 +96,8 @@ Component* Component::NewComponentOfType(const std::string &component_type) {
> >      ans = new ConvolutionComponent();
> >    } else if (component_type == "MaxpoolingComponent") {
> >      ans = new MaxpoolingComponent();
> > +  } else if (component_type == "Maxpooling3dComponent") {
>
> @danpovey <https://github.com/danpovey> I don't know of any recipes which
> are using the MaxpoolingComponent in nnet3. We are already printing
> deprecation warning for Convolution1DComponent. How about removing both
> Convolution1DComponent and MaxpoolingComponent in this commit ? I think it
> might be better to use the name MaxpoolingComponent instead of
> Maxpooling3dComponent for the new component.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/460/files#r50627193>.
>